src/runtime/cache.ts

```diff
@@ -142,14 +139,16 @@ export function defineCachedEventHandler<T = any>(
 ): EventHandler<T> {
   const _opts: CacheOptions<ResponseCacheEntry<T>> = {
     ...opts,
-    getKey: (event) => {
-      const url = event.req.originalUrl || event.req.url;
-      const friendlyName = decodeURI(parseURL(url).pathname)
-        .replace(/[^\dA-Za-z]/g, "")
-        .slice(0, 16);
-      const urlHash = hash(url);
-      return `${friendlyName}.${urlHash}`;
-    },
+    getKey:
+      opts.getKey ||
+      ((event) => {
+        const url = event.req.originalUrl || event.req.url;
```

Comment:
Maybe we can call `getKey` inside this function? this way if it returns null, we can fallback to the default key behavior.

We also need to escape invalid chars also in all conditions (including `..`) as they can lead to security issues.

Reply:
Yes, I think it makes sense to call it inside and fallback to the default; as for the escaping of invalid chars, do you mean escaping them from the returned key?

Reply:
Yes, to only allow ascii charachters like later regex we have.